src/test/java/org/assertj/core/internal/files/Files_assertIsDirectoryRecursivelyContaining_SyntaxAndPattern_Test.java

```diff
@@ -0,0 +1,138 @@
+ */
+package org.assertj.core.internal.files;
+
+import org.assertj.core.api.AssertionInfo;
+import org.assertj.core.internal.Files;
+import org.assertj.core.internal.FilesSimpleBaseTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import static java.lang.String.format;
+import static org.assertj.core.error.ShouldBeDirectory.shouldBeDirectory;
+import static org.assertj.core.error.ShouldContain.directoryShouldContain;
+import static org.assertj.core.internal.Files.toAbsolutePaths;
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.assertj.core.util.Lists.emptyList;
+import static org.assertj.core.util.Lists.list;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Tests for <code>{@link Files#assertIsDirectoryRecursivelyContaining(AssertionInfo, File, String)}</code>
+ *
+ * @author David Haccoun
+ */
+public class Files_assertIsDirectoryRecursivelyContaining_SyntaxAndPattern_Test extends FilesSimpleBaseTest {
+
+  private static final String TXT_EXTENSION_PATTERN = "regex:.+\\.txt";
+  private static final String TXT_EXTENSION_PATTERN_DESCRIPTION = format("the '%s' pattern",
+                                                                         TXT_EXTENSION_PATTERN);
+
+  @ParameterizedTest
+  @ValueSource(strings = { "regex:.+oo2\\.data", "regex:.+\\.json", "regex:.+bar2\\.json" })
+  void should_pass_if_actual_contains_one_file_matching_the_given_pathMatcherPattern(String pattern) {
+    // GIVEN
+    createDefaultFixturePaths();
+    // WHEN-THEN
+    files.assertIsDirectoryRecursivelyContaining(INFO, tempDirAsFile, pattern);
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = { "regex:.+\\.data", "regex:.+foobar.*", "regex:.+root.+foo.*" })
+  void should_pass_if_actual_contains_some_files_matching_the_given_pathMatcherPattern(String pattern) {
+    // GIVEN
+    createDefaultFixturePaths();
+    // WHEN-THEN
+
+    files.assertIsDirectoryRecursivelyContaining(INFO, tempDirAsFile, pattern);
+  }
+
+  private void createDefaultFixturePaths() {
+    // @format:off
+    // The layout :
+    //  /root/
+    //       /root/foo/
+    //       /root/foo/foobar
+    //       /root/foo/foobar/foobar1.data
+    //       /root/foo/foobar/foobar2.json
+    //       /root/foo2.data
+    // @format:on
+    Path rootDir = createDirectoryWithDefaultParent("root", "foo2.data");
+    Path fooDir = createDirectory(rootDir, "foo");
+    createDirectory(fooDir, "foobar", "foobar1.data", "foobar2.json");
+  }
+
+  @Test
+  void should_pass_if_all_actual_files_matching_the_given_pathMatcherPattern() {
+    Path fooDir = createDirectory(tempDir, "foo", "foo2.data");
+    createDirectory(fooDir, "foo3");
+    files.assertIsDirectoryRecursivelyContaining(INFO, tempDirAsFile, "regex:.*foo.*|.*tmp");
```

Comment:
add GIVEN WHEN THEN (for consistency more than anything)

Reply:
done